### DISCUSSION TITLE
[SOLVED] What affects this part of the feed GUI?

### DISCUSSION THREAD

**Discussion by God-damnit-all** [martinrotter/rssguard#547] [Q&A] [answered]
I've been working on setting up some of my own feeds using scripts, but I can't figure out what in the output affects this part of the GUI.

![image](https://user-images.githubusercontent.com/53661808/143608314-20994769-20fe-4f69-b8ee-3e828637c517.png)

**Comment by martinrotter** [comment, marked as answer]
that is just source code HTML/CSS of your article

btw, there is WIP on new default light/dark skin

https://github.com/martinrotter/rssguard/pull/540

**Reply by akinokonomi** [reply]
At the same time, it shows the base URL of the website, if the feed was fetched online:

![image](https://user-images.githubusercontent.com/91767492/143611244-f9b012d5-395d-4f15-9db3-e3feef4f7ea9.png)

I've tested it just now: Copied the part of real feed, then `cat`ed it to RSS Guard, and I see the same as @ImportTaste.
![image](https://user-images.githubusercontent.com/91767492/143611962-a3d644fe-8e00-4a67-8a7f-54830a9d0e53.png)
(This is a part of my JS from code.)

Why so?

**Reply by akinokonomi** [reply]
I don't use message viewer toolbars, so this is the first time I'm seeing it.

**Reply by God-damnit-all** [reply]
> that is just source code HTML/CSS of your article

That's a weird place to put it. Thanks for the info!

**Reply by martinrotter** [reply]
I do not do it, it's just how the web component works, I will look into it.

**Reply by martinrotter** [reply]
> 

If RSS Guard can deduce "base" URL of article, it then sets this URL as active. This is because many feeds use relative URLs for images etc. which work only if base URL is set. If base URL cannot be deduced, then no URL is set in which case web component forces this "HTML" address.

**Reply by God-damnit-all** [reply]
> > 
> 
> If RSS Guard can deduce "base" URL of article, it then sets this URL as active. This is because many feeds use relative URLs for images etc. which work only if base URL is set. If base URL cannot be deduced, then no URL is set in which case web component forces this "HTML" address.

@martinrotter The feed's Source is set to Script, so that would explain why it isn't obtaining a base URL.

**Comment by akinokonomi** [comment]
But the URLs, and everything else expected to be exactly the same, since in my case I simply put downloaded feed file of that same feed.

This how it goes:
1. For some millisecond it shows `http://rssguard.blank/`
2. Here is either base site's URL (in case of proper feed) appear, or HTML page `data:text/html;charset=UTF-8,%3C%21DOCTYPE`-and-so-on (in case of script)

I don't have that many scripts, but this is how it is with every of them. 
And I've seen no feed which would show `data:text/html;charset=UTF-8,%3C%21DOCTYPE` instead of base URL.

**Reply by akinokonomi** [reply]
So, I think that the base URL is not deduced from _article's_ URL, but from _feed's_ URL which is passed to RSS Guard through _Source_ input box in "Add new feed" modal.

**Reply by God-damnit-all** [reply]
> But the URLs, and everything else expected to be exactly the same, since in my case I simply put downloaded feed file of that same feed.
> 
> This how it goes:
> 
> 1. For some millisecond it shows `http://rssguard.blank/`
> 2. Here is either base site's URL (in case of proper feed) appear, or HTML page `data:text/html;charset=UTF-8,%3C%21DOCTYPE`-and-so-on (in case of script)
> 
> I don't have that many scripts, but this is how it is with every of them. And I've seen no feed which would show `data:text/html;charset=UTF-8,%3C%21DOCTYPE` instead of base URL.

It happens when Script is made the Source of a feed.

**Reply by martinrotter** [reply]
> 

That is correct.

**Reply by God-damnit-all** [reply]
> > 
> 
> That is correct.

Do you think you could add a 'base URL' override? This wouldn't be only useful for Script RSS feeds, I've noticed some RSS feeds using relative paths that don't match up with where the RSS feed is located on the remote server.

**Reply by akinokonomi** [reply]
> It happens when Script is made the Source of a feed.

Of course, I didn't even question this matter.

[Here](https://github.com/martinrotter/rssguard/discussions/547#discussioncomment-1707446) @martinrotter said
> If RSS Guard can deduce "base" URL of article, it then sets this URL as active.

But the base URL is not deduced from _article_, it is likely from _feed_. (And because RSS Guard cannot "deduce" base URL  from the script path, the source HTML is shown there.)
